test/shacheck.c

```diff
@@ -78,24 +90,25 @@ int main (int argc, char *argv[]) {
     } else {
         waitpid(child_pid, &status, 0);
     }
+#endif
     if (status != 0) {
         printf("Error running command\n");
         exit(1);
     }
 
     /* Open zchunk file and check that checksum matches */
-    int in = open(outf, O_RDONLY);
+    int in = open(outf, O_RDONLY | O_BINARY);
     if(in < 0) {
         perror("");
         printf("Unable to open %s for reading", outf);
         exit(1);
     }
     /* Files must be smaller than 1MB  */
-    char data[1024*1024] = {0};
+    char* data = malloc(1024*1024);
```

Comment:
Since we're now getting this memory off the heap, can you please validate that we've actually been allocated the memory, please?

Reply:
Done :)